misk/src/test/kotlin/misk/testing/PortPicker.kt

```diff
@@ -0,0 +1,8 @@
+package misk.testing
+
+import java.net.ServerSocket
+
+/** Finds a free port for testing */
```

Comment:
Ick. MockWebServer’s approach to this is much simpler. It does exactly this; but it actually uses the port it reserved for its server socket

Reply:
(at a minimum this is racy)

Reply:
... and now I’m wondering whether Jetty has a similar facility

Reply:
Jetty does not - there's no way to interrogate it for the bound port (I looked for this before adding PortPicker)

Reply:
Also we don't directly expose the bound jetty instance or jetty service, but I can probably change that with some Guice-fu